test/functional/rpc_generateblock.py

```diff
@@ -5,25 +5,43 @@
 '''Test generateblock rpc.
 '''
 
+from decimal import Decimal
 from test_framework.test_framework import BitcoinTestFramework
+from test_framework.address import ADDRESS_BCRT1_P2WSH_OP_TRUE
 from test_framework.util import (
     assert_equal,
     assert_raises_rpc_error,
 )
+from test_framework.messages import (
+    COIN,
+    COutPoint,
+    CTransaction,
+    CTxIn,
+    CTxInWitness,
+    CTxOut,
+)
+from test_framework.script import (
+    CScript,
+    OP_TRUE,
+)
+from test_framework.util import (
+    satoshi_round,
+    hex_str_to_bytes,
+)
+from test_framework.wallet import MiniWallet
 
 
 class GenerateBlockTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 1
 
-    def skip_test_if_missing_module(self):
-        self.skip_if_no_wallet()
-
     def run_test(self):
         node = self.nodes[0]
+        miniwallet = MiniWallet(node)
+        address = ADDRESS_BCRT1_P2WSH_OP_TRUE
+        scriptPubKey = hex_str_to_bytes(node.validateaddress(ADDRESS_BCRT1_P2WSH_OP_TRUE)['scriptPubKey'])
 
         self.log.info('Generate an empty block to address')
-        address = node.getnewaddress()
         hash = node.generateblock(output=address, transactions=[])['hash']
```

Comment:
I now realize I have wrote this in a very misleading way. Because even though this is getting generated to the `miniwallet` address I don't append it to `self._utxos` so therefore not spendable by `send_self_transfer()`. My thought was to instead generate to the unspendable address (ADDRESS_BCRT1_UNSPENDABLE) to make it clear its not spendable by `miniwallet` and then generate 11 blocks instead of 10 to `miniwallet` to have sufficient blocks for the rest of the test. Thoughts?